thespian/system/multiprocCommon.py

```diff
@@ -544,24 +565,7 @@ def startChild(childClass, globalName, endpoint, transportClass,
     # logging.shutdown() because (a) that does not do enough to reset,
     # and (b) it shuts down handlers, but we want to leave the parent's
     # handlers alone.
-    if logDefs:
-        levelIn = lambda d: d.get('level', 0)
-        minLevelIn = lambda l: min(list(l)) if list(l) else 0
-        levels = list(
-            filter(None,
-                   ([ minLevelIn([levelIn(logDefs[key][subkey])
-                                  for subkey in logDefs[key]
-                                  # if subkey in [ 'loggers', 'handlers'
-                                  if isinstance(logDefs[key][subkey], dict)
-                   ])
-                      if key in ['loggers', 'handlers'] and isinstance(logDefs[key], dict)
-                      else (levelIn(logDefs[key]) if key == 'root'
-                            else (logDefs[key] if key == 'level'
-                                  else None))
```

Comment:
Lines 558-560 are handling two cases not shown in the Thespian/logsetup.py:

    logcfg = { 'version': 1,
               ...,
               'root' : { 'level': N, ... },
               'level': N,
    }

The Python dictconfig logging schema (https://docs.python.org/3/library/logging.config.html#dictionary-schema-details) handles a 'level' specification at the top, and also for the root logger.

Your updated version will need to handle those as well, along with the case where there is no level specified anywhere (the `filter(None, [... else None])` in the original.

Now that you've added doctests (thanks!) you could add more doctest cases for these scenarios to make sure we capture them all in more clarity than my original complicated+obscure implementation had.

Reply:
corrected and more tests added.